src/rust/src/x509/certificate.rs

```diff
@@ -731,6 +731,103 @@ pub(crate) fn parse_access_descriptions(
     Ok(ads.to_object(py))
 }
 
+fn parse_naming_authority<'p>(
+    py: pyo3::Python<'p>,
+    authority: NamingAuthority<'p>,
+) -> CryptographyResult<pyo3::PyObject> {
+    let py_id = match &authority.id {
+        Some(data) => oid_to_py_oid(py, data)?.to_object(py),
+        None => py.None(),
+    };
+    let py_url = match authority.url {
+        Some(data) => pyo3::types::PyString::new_bound(py, data.as_str()).to_object(py),
+        None => py.None(),
+    };
+    let py_text = match authority.text {
+        Some(data) => parse_display_text(py, data)?,
+        None => py.None(),
+    };
+
+    Ok(types::NAMING_AUTHORITY
+        .get(py)?
+        .call1((py_id, py_url, py_text))?
+        .to_object(py))
+}
+
+fn parse_profession_infos<'a>(
+    py: pyo3::Python<'_>,
+    profession_infos: &asn1::SequenceOf<'a, ProfessionInfo<'a>>,
+) -> CryptographyResult<pyo3::PyObject> {
```

Comment:
This should return a `Bound<'p, PyAny>`, in general our habbit of returning `PyObject` all over hte place was a mistake we don't want to extend.

Reply:
@alex done - I also went through other functions I added (`parse_naming_authority` and `parse_admissions`) and adjusted them as well; this also applies to your other suggestions. Please verify that I did what you indeed had in mind!